### ISSUE TITLE
hf mf csetuid

### ISSUE THREAD

**Issue by iceman1001** [Proxmark/proxmark3#35]
For a long while my "hf mf csetuid" has been all but reliable. 
With Piwi's changes to iso14443a_select_card, I was told about the critical timings which are now in order.
I also noticed some entries on the forum asking about the magic cards not working.
When I commented out some "spindelays" here,  https://github.com/Proxmark/proxmark3/blob/master/armsrc/mifarecmd.c#L906 ,   I got my "hf mf csetuid" to work without a problem.

I suspect this is the problem with the other magic-chines commands too.

I suggest the patch to comment out  row 906 to 909. 

Please review and come back with feedback.

**Comment by iceman1001** [comment]
If someelse than me can verify that with the following patch to this issue "hf mf csetuid"  working. It works for me but it would be nice to have it a bit more tested.

If ok, then this issue can be closed.
 
commit: https://github.com/Proxmark/proxmark3/commit/3fe4ff4f0329d6bde9585b77966d42dfc3f612f5

**Comment by iceman1001** [comment]
However I noticed something yesterday again..  And read about it on the forum in some thread a year back.  http://www.proxmark.org/forum/viewtopic.php?pid=7883#p7883


It is the default behavior of this commad,  "hf mf csetuid"  which is to change the uid on a magic s50 tag.  There is two behaviors which I want to change.   Internally the "csetuid" issues a "hf mf csetblk" command with a block 0 with new UID and SAK&ATAQ,

A) When you use the command, it resets not only the uid in Block0, but it also clears the rest of the block0 with zeros.   It makes your cloned copy not an exact copy anymore.

B) The bytes[5-6-7] in Block 0 on a magic tag is  the ATAQ & SAK.  The normal behavior of the command is to set it to "88 00 04"  as default SAK & ATAQ...   Which also makes your cloned copy not an exact copy anymore.

Suggested solutions. 
A)  issue a readblock 0, before and copy bytes 8-16 to the new bytearray.  This leaves your block0 intact and just changes the UID.   I have implemented this solution.

B) make the SAK & ATAQ optional from command line.  Which allows for it easy to make the magic card identify itself as different kinds of tags.

Is this of interest for the community? Feedback on this?
